apprentice-react-o-shop/src/components/Auth/LoginForm.js

```diff
@@ -31,22 +35,22 @@ function LoginForm() {
 
     const inputFieldList = []
 
-    for (const key in formFields){
-        if (formFields[key].login){
-            const referenceVariable = (formFields[key].name === "username") ? username : password
+    formFields.forEach(inputObject => {
+        if (inputObject.login){
+            const referenceVariable = (inputObject.name === "username") ? username : password
```

Comment:
This can be shortened down to just `const referenceVariable = inputObject.name`. You know that value will be username or password so just set the const to that result.

Reply:
I don't think that works because we want to reference the username variable and not the string 'username' when we pass props to InputField.js